Modules/Core/Common/include/itkConstShapedNeighborhoodIterator.h

```diff
@@ -374,8 +374,15 @@ class ITK_TEMPLATE_EXPORT ConstShapedNeighborhoodIterator : private Neighborhood
   /** Add non-zero neighborhood offsets to the active list. The
    * radius of the neighborhood must match the radius of the shaped
    * iterator */
+  template <typename NeighborPixelType>
   void
-  CreateActiveListFromNeighborhood(const NeighborhoodType &);
+  CreateActiveListFromNeighborhood(const Neighborhood<NeighborPixelType, Self::Dimension> &);
+  void
+  CreateActiveListFromNeighborhood(const NeighborhoodType & neighborhood)
+  {
+    // just delegate to the templated version
+    this->CreateActiveListFromNeighborhood<>(neighborhood);
+  }
```

Comment:
Thanks @dzenanz. Would it be possible to just remove the old `CreateActiveListFromNeighborhood(const NeighborhoodType &)` now, as it seems totally redundant when we get your more generic version?

Reply:
I thought about it too, but I am not sure that all the calls that worked before would continue working. The template variant relies on template argument being properly deduced, and that sometimes does not work.

Reply:
I tried it locally, and it appeared to work just fine, after removing the old `CreateActiveListFromNeighborhood(const NeighborhoodType &)` overload.

Reply:
There is not a lot of code inside ITK which exercises this. I would rather be cautious.